packages/core/lib/segments/attributes/subsegment.js

```diff
@@ -124,15 +116,15 @@ Subsegment.prototype.addPrecursorId = function(id) {
  */
 
 Subsegment.prototype.addAnnotation = function(key, value) {
-  if (!(isBoolean(value) || isString(value) || isFinite(value))) {
+  if (!['boolean', 'number', 'string'].includes(typeof value)) {
```

Comment:
I like how you've kept the conditional compact, but if the type is a number we also want to make sure it is finite (not Infinity or NaN) using the native `isFinite` function. Something like this:

```suggestion
  if (!(typeof value === 'boolean' || typeof value === 'string' || (typeof value === 'number' && isFinite(value))) {
```